### DISCUSSION TITLE
Next JS build time

### DISCUSSION THREAD

**Discussion by adduss** [vercel/next.js#37015] [Help] [answered]
I would like to understand how next build is calculating time of particular items. When I run it `npm run build` my results are:
![Screenshot 2022-05-18 at 15 37 08](https://user-images.githubusercontent.com/5172081/169055827-7bbbef3f-06b3-4cbd-87b4-492b79cb2be5.png)

In red there is value "1592335 ms" which means that generation of 5k pages took around 26.5 minutes:
![Screenshot 2022-05-18 at 15 34 38](https://user-images.githubusercontent.com/5172081/169054829-bcf9daf0-e060-487e-8e59-84e26687e053.png)

But you can see from dates before and after build that in reality it took around 3 minutes. I started to think that maybe it is "159233**.**5 ms", can anyone please explain how to read properly build time?

Thank you for your time!

**Comment by icyJoseph** [comment, marked as answer]
Mmm I think that's the combined time of all Workers that built the pages. As in total worker time spent building pages.

I see it says avg, 317 ms, and that times 4993 (more paths) is 1582781 ms, which like 0.5% off from 1592335 ms.

I found this in the source, https://github.com/vercel/next.js/blob/b2045c7669982edbb7002c7f88f8e7665a2168f7/packages/next/export/worker.ts#L302, there's a few more on that file. And then these seem to be aggregated, as total worker time spent.

And here's the aggregate: https://github.com/vercel/next.js/blob/b2045c7669982edbb7002c7f88f8e7665a2168f7/packages/next/build/utils.ts#L190

Fun to jump around in this area of the code 👍

**Reply by adduss** [reply]
Thank you @icyJoseph for pointing me direction!

I went debugged build process and all is clear. If someone is interested how NextJS calculates number of workers it is max value from 1 and number of CPUs on your machine minus 1 - [config-shared.ts](https://github.com/vercel/next.js/blob/fe3d6b7aed5e39c19bd4a5fbbf1c9c890e239ea4/packages/next/server/config-shared.ts#L483)

It is possible to modify number of workers using next config file by:
`experimental: {
    cpus: <override_no_of_cpus>
 }`
